snuba/datasets/errors_replacer.py

```diff
@@ -206,93 +213,129 @@ def set_project_needs_final(
     ]
-
     for exclude_groups_key, _ in exclude_groups_keys_and_types:
+        # remove stale excluded groups
         p.zremrangebyscore(
             exclude_groups_key, float("-inf"), now - settings.REPLACER_KEY_TTL
         )
+        # TODO: put this into another for loop to make post processing easier
+        # get up to date excluded groups
         p.zrevrangebyscore(
             exclude_groups_key, float("inf"), now - settings.REPLACER_KEY_TTL
         )
 
+    # get the replacement types for metrics purposes
     for _, needs_final_type_key in needs_final_keys_and_type_keys:
         p.get(needs_final_type_key)
 
     for _, type_key in exclude_groups_keys_and_types:
+        # cleanup the old exclude group replacement types
         p.zremrangebyscore(type_key, float("-inf"), now - settings.REPLACER_KEY_TTL)
+        # get the new exclude group replacement types
+        # TODO: put this into another for loop to make post processing easier
         p.zrevrangebyscore(type_key, float("inf"), now - settings.REPLACER_KEY_TTL)
 
+    # retrieve the latest replaced group id's timestamp such that queries
+    # which are processing data after it, do not have to be marked as final
+    for exclude_groups_key, _ in exclude_groups_keys_and_types:
+        p.zrevrange(
+            exclude_groups_key, 0, 0, withscores=True,
+        )
+
     results = p.execute()
 
-    return process_exclude_groups_and_replacement_types_results(
+    return _process_exclude_groups_and_replacement_types_results(
         results, len(s_project_ids)
     )
 
 
-def process_exclude_groups_and_replacement_types_results(
+def _process_exclude_groups_and_replacement_types_results(
     results: List[Any], len_projects: int
-) -> Tuple[bool, Sequence[int], Set[str]]:
+) -> ProjectsQueryFlags:
     """
     Helper function for `get_projects_query_flags`.
-    `results` is in the form:
+    Given raw redis result output, return something that makes sense
+
+    `results` is a flat list of all the redis call results of get_projects_query_flags
     [
-        needs_final...,
-        exclude_groups...,
-        needs_final_replacement_types...,
-        groups_replacement_types...
+        needs_final: Sequence[timestamp]...,
+        exclude_groups: Sequence[..(num_removed_elements(don't care), List[group_id] )]...,
+        needs_final_replacement_types: Sequece[Optional[str]]...,
+        groups_replacement_types: Sequence[..(num_removed, str)]...,
+        latest_exclude_groups_replacements: Sequence[Optional[Tuple[group_id, datetime]]]...
     ]
     - `needs_final` slice is `len_projects` long
     - `excludes_groups` slice is `len_projects * 2` long
     - `needs_final_replacement_types` slice is `len_projects` long
     - `groups_replacement_types` slice is `len_projects * 2` long
+    - `latest_exclude_groups_replacements` slice is `len_projects` long
+
+    [needs_final   |exclude_groups    |...]
+    [ len(projects)|len(projects) * 2 |   ]
```

Comment:
either remove this explanation or replace the above one with what it actually is